platform/src/main/java/org/stellar/anchor/platform/paymentobserver/StellarPaymentObserver.java

```diff
@@ -0,0 +1,123 @@
+package org.stellar.anchor.platform.paymentobserver;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import org.stellar.anchor.util.Log;
+import org.stellar.sdk.KeyPair;
+import org.stellar.sdk.Server;
+import org.stellar.sdk.requests.EventListener;
+import org.stellar.sdk.requests.PaymentsRequestBuilder;
+import org.stellar.sdk.requests.SSEStream;
+import org.stellar.sdk.responses.operations.OperationResponse;
+import org.stellar.sdk.responses.operations.PaymentOperationResponse;
+import shadow.com.google.common.base.Optional;
+import shadow.com.google.gson.Gson;
+
+public class StellarPaymentObserver {
+  final Server server;
+  final List<PaymentListener> observers;
+  final List<String> accounts;
+  final List<SSEStream<OperationResponse>> streams;
+  final PageTokenStore pageTokenStore;
+
+  StellarPaymentObserver(
+      String horizonServer,
+      List<String> accounts,
+      List<PaymentListener> observers,
+      PageTokenStore pageTokenStore) {
+    this.server = new Server(horizonServer);
+    this.observers = observers;
+    this.accounts = accounts;
+    this.streams = new ArrayList<>(accounts.size());
+    this.pageTokenStore = pageTokenStore;
+  }
+
+  /** Start watching the accounts. */
+  public void start() {
+    for (String account : accounts) {
+      SSEStream<OperationResponse> stream = watch(account);
+      this.streams.add(stream);
+    }
+  }
+
+  /** Graceful shutdown. */
+  public void shutdown() {
+    for (SSEStream<OperationResponse> stream : streams) {
+      stream.close();
+    }
+  }
+
+  public SSEStream<OperationResponse> watch(String account) {
+    PaymentsRequestBuilder paymentsRequest = server.payments().forAccount(account);
```

Comment:
💡  We should configure the streamer with `limit=200` to make the requests more efficient, and `order=desc` to make sure we're streaming from oldest to newest. I think the default order is `desc`, but I suggest making it explicit here _just in case_.